util/sample_collections.py

```diff
@@ -80,13 +100,21 @@ def __enter__(self):
     def add(self, sample):
         def to_bytes(n):
             return n.to_bytes(INT_SIZE, BIG_ENDIAN)
+        if not isinstance(sample, Sample):
+            raise TypeError('provided sample is not of type Sample')
+        if self.labeled and not isinstance(sample, LabeledSample):
+            raise TypeError('writing a labeled sample collection, but provided sample is not of type LabeledSample')
```

Comment:
These `isinstance` checks break duck typing. Either check `hasattr(sample, 'transcript')` or just let it explode with an AttributeError.

Reply:
I think I'll let it explode...